common-features/data-binding/descriptors.md

```diff
@@ -0,0 +1,322 @@
+             Groupable="true"
+             FilterMode="@GridFilterMode.FilterMenu"
+             Pageable="true" PageSize="15"
+             Height="400px">
+    <GridToolBarTemplate>
+        <GridSearchBox />
+    </GridToolBarTemplate>
+    <GridColumns>
+        <GridColumn Field="@nameof(SampleData.Id)" />
+        <GridColumn Field="@nameof(SampleData.Name)" />
+        <GridColumn Field="@nameof(SampleData.Team)" />
+        <GridColumn Field="@nameof(SampleData.HireDate)" />
+    </GridColumns>
+</TelerikGrid>
+
+
+@code {
+    private MarkupString ConsoleSim { get; set; } // to showcase what you get
+
+    private async Task OnReadHandler(GridReadEventArgs args)
+    {
+        string output = string.Empty;
+
+        //get the filtering and searching criteria
+        output += "<span><strong>FILTERS:</strong><span></br>";
+        foreach (var item in args.Request.Filters)
+        {
+            if (item is CompositeFilterDescriptor)
+            {
+                CompositeFilterDescriptor currFilter = (CompositeFilterDescriptor)item;
+                output += $"START nested filter: Logical operator: {currFilter.LogicalOperator}, details:<br />";
+                foreach (FilterDescriptor nestedFilter in currFilter.FilterDescriptors)
+                {
+                    output += $"Filtered field: {nestedFilter.Member}, Filter operator: {nestedFilter.Operator}, Filter value: {nestedFilter.Value}<br />";
+                }
+                output += "END nested filter<br />";
+            }
+        }
+
+        //get the sorting criteria
+        output += "<span><strong>SORTS:</strong><span></br>";
+        foreach (SortDescriptor item in args.Request.Sorts)
+        {
+            output += $"Sorted field: {item.Member}, Sort direction: {item.SortDirection} <br />";
+        }
+
+        //get the grouping criteria
+        output += "<span><strong>GROUPS:</strong><span></br>";
+        foreach (GroupDescriptor item in args.Request.Groups)
+        {
+            output += $"Grouped field: {item.Member}, Group sort direction: {item.SortDirection} <br />";
+        }
+
+        ConsoleSim = new MarkupString(output);
+
+        var result = PristineData.ToDataSourceResult(args.Request);
+
+        args.Data = result.Data;
+        args.Total = result.Total;
+    }
+
+    private IEnumerable<SampleData> PristineData = Enumerable.Range(1, 300).Select(x => new SampleData
+        {
+            Id = x,
+            Name = "name " + x,
+            Team = "team " + x % 5,
+            HireDate = DateTime.Now.AddDays(-x).Date
+        });
+
+    public class SampleData
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public string Team { get; set; }
+        public DateTime HireDate { get; set; }
+    }
+}
+````
+
+## Example with Component State
+
+You can obtain the FilterDescriptor, SearchFilter, SortDescriptor, GroupDescriptor through the component state.
+
+````CSHTML
```

Comment:
Make sure that all comments that represent complete sentences use the appropriate capitalization and punctuation.